### PR TITLE
[SPARK-46022][SPARK-46017][DOCS][PYTHON] Remove deprecated function APIs from documents

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to remove deprecated functions API from documents.

This PR also fixes SPARK-46017 that was the known issue on Mac OS when building the PySpark docs below:
```
$ make clean html
...
reference/pyspark.ss/api/pyspark.sql.streaming.StreamingQueryManager.removeListereading sources... [ 98%] reference/pyspark.ss/api/pyspark.sql.streaming.StreamingQueryManager.resetTerminreading sources... [100%] user_guide/sql/type_conversions
...
Warning, treated as error:
/.../spark/python/docs/source/reference/pyspark.sql/functions.rst.rst:223:autosummary: stub file not found 'pyspark.sql.functions.shiftleft'. Check your autosummary_generate setting.
make: *** [html] Error 2
```
### Why are the changes needed?

To do not encourage using deprecated APIs by hiding them from API reference docs.

### Does this PR introduce _any_ user-facing change?

No, it's documentation update.

### How was this patch tested?

Manually built docs on Mac OS:

Screenshot 2023-11-21 at 6 55 25 PM

The existing CI should pass.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/docs/source/reference/pyspark.sql/functions.rst [hunk 1]
```diff
@@ -504,18 +496,6 @@ Generator Functions
     stack
 
 
-Partition Transformation Functions
-----------------------------------
-.. autosummary::
-    :toctree: api/
-
-    partitioning.years
```
Comment (HyukjinKwon) [inline python/docs/source/reference/pyspark.sql/functions.rst:512]: ?? this is not deprecated.

**Review by HyukjinKwon — APPROVED** [review]
LGTM otherwise. I face the same issue either.

File: python/docs/source/reference/pyspark.sql/functions.rst (hunk 1 above)
Comment (dongjoon-hyun) [reply python/docs/source/reference/pyspark.sql/functions.rst:512]: Ya, these are added recently.

Comment (itholic) [reply python/docs/source/reference/pyspark.sql/functions.rst]: Ohhh,,, I was confused with `years`. Just revert the change. Thanks!!

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**